packages/slate/src/transforms/node.ts

```diff
@@ -716,7 +716,9 @@ export const NodeTransforms = {
 
       const rangeRef = Range.isRange(at) ? Editor.rangeRef(editor, at) : null
       const matches = Editor.nodes(editor, { at, match, mode, voids })
-      const pathRefs = Array.from(matches, ([, p]) => Editor.pathRef(editor, p))
+      const pathRefs = Array.from(matches, ([, p]) =>
+        Editor.pathRef(editor, p)
+      ).reverse()
```

Comment:
```suggestion
      // unwrapNode will call liftNode which does not support splitting the mode when nested.
      // If we do not reverse the order and call it from top to the bottom, it will remove all blocks 
      // that wrap target node. So we reverse the order.
      ).reverse()
```

I think a comment helps here. I'm happy to be told this is unnecessary. :)